src/main/java/graphql/validation/constraints/standard/PatternConstraint.java

```diff
@@ -44,28 +47,39 @@ public Documentation getDocumentation() {
     @Override
     public boolean appliesToType(GraphQLInputType inputType) {
         return isOneOfTheseTypes(inputType,
-                Scalars.GraphQLString
-        );
+            Scalars.GraphQLString) || isList(inputType);
     }
 
     @Override
     protected List<GraphQLError> runConstraint(ValidationEnvironment validationEnvironment) {
         Object validatedValue = validationEnvironment.getValidatedValue();
+        GraphQLInputType argumentType = validationEnvironment.getValidatedType();
+
         if (validatedValue == null) {
             return emptyList();
         }
-        String strValue = String.valueOf(validatedValue);
 
-        GraphQLDirective directive = validationEnvironment.getContextObject(GraphQLDirective.class);
+        List<Object> validatedValues;
+
+        if (isList(argumentType)) {
+            validatedValues = (ArrayList)validatedValue;
```

Comment:
Cast as just `(List)` and not the specific type please

Reply:
Certainly - update made